presto-spi/src/main/java/com/facebook/presto/spi/RecordPageSource.java

```diff
@@ -126,7 +126,12 @@ else if (javaType == double.class) {
                         }
                         else if (javaType == Slice.class) {
                             Slice slice = cursor.getSlice(column);
-                            type.writeSlice(output, slice, 0, slice.length());
+                            if (slice == null) {
```

Comment:
when does this case happen?

Reply:
OK, I managed to fix this so it no longer needs this extra null checks and changes are limited to the hive package.